bin/add_hypothetical_protein_descriptions.py

```diff
@@ -0,0 +1,714 @@
+                            found_function, function_source = get_function(
+                                attributes_dict["ID"],
+                                attributes_dict,
+                                eggnog_info,
+                                ipr_info,
+                                ipr_memberdb_only,
+                            )
+
+                            if not function_source == "UniFIRE":
+                                found_function = clean_up_function(found_function)
+                                if function_source == "eggNOG":
+                                    found_function, function_source, attributes_dict = (
+                                        keep_or_move_to_note(
+                                            found_function,
+                                            function_source,
+                                            attributes_dict,
+                                        )
+                                    )
+                            found_function = escape_reserved_characters(found_function)
+                            attributes_dict["product"] = found_function
+                            attributes_dict = insert_product_source(
+                                attributes_dict, function_source
+                            )
+                        else:
+                            attributes_dict = insert_product_source(
+                                attributes_dict, "Prokka"
+                            )
+                        col9_updated = update_col9(attributes_dict)
+                        file_out.write(
+                            "\t".join(
+                                [
+                                    contig,
+                                    tool,
+                                    feature,
+                                    start,
+                                    end,
+                                    blank1,
+                                    strand,
+                                    blank2,
+                                    col9_updated,
+                                ]
+                            )
+                            + "\n"
+                        )
+                    else:
+                        file_out.write(line)
+                else:
+                    file_out.write(line)
+            else:
+                file_out.write(line)
+
+
+def keep_or_move_to_note(found_function, function_source, col9_dict):
+    """
+    Function aims to identify if a description is likely to be a sentence/paragraph rather than
+    a succinct function description. If it's the former, move it to note and revert function to
+    "hypothetical protein".
+    """
+    move_to_note = False
+    description_length = len(found_function.split())
+    if description_length < 12:  # likely to be a proper description
+        text_to_avoid = [
```

Comment:
I would push there 2 constants to the top of the file and use upper case  (it's a convention for constants)

Reply:
thanks, changed that